apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -348,6 +348,37 @@ func (p *ProvisionerAPI) Series(args params.Entities) (params.StringResults, err
 	return result, nil
 }
 
+// AvailabilityZone returns a provider-specific availability zone for each given machine entity
+func (p *ProvisionerAPI) AvailabilityZone(args params.Entities) (params.StringResults, error) {
```

Comment:
This needs a test.

Reply:
https://github.com/juju/juju/pull/7795/commits/fb986bbbad2c377610760f883c76297eab2655fb